src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -141,6 +141,34 @@ export class FocusNextTerminalAction extends Action {
 	}
 }
 
+export class FocusTerminalByNumberAction extends Action {
+	private static ID_PREFIX = 'workbench.action.terminal.focus';
+
+	constructor(
+		id: string, label: string,
+		@ITerminalService private terminalService: ITerminalService
+	) {
+		super(id, label);
+	}
+
+	public run(event?: any): TPromise<any> {
+		this.terminalService.setActiveInstanceByIndex(this.getTerminalNumber() - 1);
+		return this.terminalService.showPanel(true);
+	}
+
+	public static getId(n: number): string {
+		return FocusTerminalByNumberAction.ID_PREFIX + n;
+	}
+
+	public static getLabel(n: number): string {
+		return nls.localize(`workbench.action.terminal.focusByNumber${n}`, `Focus Terminal ${n}`, n);
```

Comment:
Almost there 😃 I think this should be:

```ts
return nls.localize('workbench.action.terminal.focusByNumber', 'Focus Terminal {0}', n);
```

That way our localization system will understand it.